macro/src/main/scala-2.x/monocle/macros/syntax/MacroSyntax.scala

```diff
@@ -76,4 +108,27 @@ class MacroAsOpsImpl(val c: blackbox.Context) {
       q"""$subj.andThen(_root_.monocle.macros.GenPrism[${c.weakTypeOf[To]}, ${c.weakTypeOf[CastTo]}])"""
     )
   }
+
+  def field_impl[Optic[_, _], From, To: c.WeakTypeTag, Next](lambda: c.Expr[To => Next]): c.Expr[Optic[From, Next]] = {
+    import c.universe._
+
+    val subj = c.prefix.tree match {
+      case Apply(TypeApply(_, _), List(x)) => x
+      case t                               => c.abort(c.enclosingPosition, s"Invalid prefix tree ${show(t)}")
+    }
+
+    lambda match {
```

Comment:
This duplicate code from GenLens macro, I couldn't find out how to pass a Boolean flag ...

I don't think it matters too much since it is for Scala 2 and we don't plan to touch macro further in this version of Scala.